File: Minimum Flips to Make a OR b Equal to c/Minimum Flips to Make a OR b Equal to c.cpp
class Solution
{
public:
    int minFlips(int a, int b, int c)
    {
        int ctr = 0;
        while (c || a || b)
        {
            if (c & 1)
            {
                ctr += !(a & 1) && !(b & 1);
            }
            else
            {
                ctr += (a & 1) + (b & 1);
            }
            a >>= 1, b >>= 1, c >>= 1;
        }
        return ctr;
    }
};

File: Minimum Flips to Make a OR b Equal to c/Minimum Flips to Make a OR b Equal to c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        ctr = 0
        while c != 0 or a != 0 or b != 0:
            if c % 2 != 0:
                ctr += 1 if (a%2==0) and (b%2==0) else 0
            else:
                ctr += 1 if a%2!=0 else 0
                ctr += 1 if b%2!=0 else 0
            a//=2
            b//=2
            c//=2
        return ctr